drivers/serial/uart_nrf_sw_lpuart.c

```diff
@@ -591,17 +832,60 @@ static int lpuart_init(struct device *dev)
 	k_timer_init(&data->tx_timer, tx_timeout, NULL);
 	k_timer_user_data_set(&data->tx_timer, dev);
 
-	return uart_callback_set(data->uart, uart_callback, (void *)dev);
+	err = uart_callback_set(data->uart, uart_callback, (void *)dev);
+	if (err < 0) {
+		return -EINVAL;
+	}
+
+	if (IS_ENABLED(CONFIG_NRF_SW_LPUART_INT_DRIVEN)) {
+		err = uart_callback_set(dev, int_driven_evt_handler, dev);
+		if (err < 0) {
+			return -EINVAL;
+		}
+
+		err = api_rx_enable(dev, data->int_driven->rxbuf,
+				    sizeof(data->int_driven->rxbuf), 1);
+	}
+
+	data->txbyte = -1;
+
+	return err;
 }
 
 static int api_poll_in(struct device *dev, unsigned char *p_char)
 {
-	return -ENOTSUP;
+	if (IS_ENABLED(CONFIG_NRF_SW_LPUART_INT_DRIVEN)) {
+		return api_fifo_read(dev, p_char, 1) ? 0 : -1;
+	}
+
+	return -EBUSY;
```

Comment:
API documentation says:
>-EBUSY If reception was enabled using uart_rx_enabled"

Thus, `-ENOTSUP` seems to suit better here.

Reply:
done